include/sys_clock.h

```diff
@@ -109,11 +109,10 @@ static ALWAYS_INLINE s32_t z_ms_to_ticks(s32_t ms)
 #ifdef CONFIG_SYS_CLOCK_EXISTS
 
 #ifdef _NEED_PRECISE_TICK_MS_CONVERSION
-	int cyc = sys_clock_hw_cycles_per_sec();
-
 	/* use 64-bit math to keep precision */
-	return (s32_t)ceiling_fraction((s64_t)ms * cyc,
-		((s64_t)MSEC_PER_SEC * cyc) / CONFIG_SYS_CLOCK_TICKS_PER_SEC);
+	return (s32_t)ceiling_fraction(
+		(s64_t)ms * sys_clock_hw_cycles_per_sec(),
+		(s64_t)MSEC_PER_SEC * sys_clock_hw_cycles_per_tick());
```

Comment:
The `sys_clock_hw_cycles_per_tick()` looks suspicious, there are no ticks mentioned in other parts of this calculation.

Reply:
The `sys_clock_hw_cycles_per_sec() / sys_clock_hw_cycles_per_tick` is the real number of ticks per second (if you take under account fraction part of the division). So to get number of ticks for given number of miliseconds, we have to multiply this value by `ms / MSEC_PER_SEC`.